.github/workflows/tests-npgsql.yml

```diff
@@ -0,0 +1,73 @@
+name: Npgsql Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    name: ".NET: ${{ matrix.dotnet-version }}
+     Npgsql: ${{ matrix.npgsql-version }}
+     CrateDB: ${{ matrix.cratedb-version }}
+     on ${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]  # , macos-latest
+        dotnet-version: ['3.1.x', '5.0.x']  # 6.0.0-preview.4
+        cratedb-version: ['4.3.2', '4.4.3', '4.5.1', 'nightly']
+        npgsql-version: ['5.0.5', '6.0.0-preview4']
+        #include:
+        #- dotnet-version: 3.1.x
+        #  dotnet-framework: netcoreapp3.1
+        #- dotnet-version: 5.0.x
+        #  dotnet-framework: net5.0
+      fail-fast: true
+
+    # https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers
+    services:
+      cratedb:
+        image: crate/crate:${{ matrix.cratedb-version }}
+        ports:
+          - 4200:4200
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up .NET ${{ matrix.dotnet-version }}
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet-version }}
+
+      - name: Invoke tests on netcoreapp3.1
+        if: matrix.dotnet-version == '3.1.x'
+        run: |
+          cd spikes/npgsql-vanilla
+          dotnet test --framework=netcoreapp3.1
```

Comment:
Wouldn't something like `dotnet test --framework=${{ matrix.dotnet-version }}` work to avoid the conditionals?

Reply:
Unfortunately not, because we would need a map to associate the .NET framework identifier used by GHA's `actions/setup-dotnet` (3.1.x, 5.0.x) to the identifier used by the `dotnet` command's argument `--framework` (netcoreapp3.1, net5.0).

I tried to do it with the `include:` attribute on GHA's matrix definition but failed on that.